pkg/apis/serving/fieldmask.go

```diff
@@ -528,10 +530,39 @@ func ResourceRequirementsMask(in *corev1.ResourceRequirements) *corev1.ResourceR
 
 }
 
+// PodSecurityContext performs a _shallow_ copy of the Kubernetes PodSecurityContext object into a new
+// Kubernetes PodSecurityContext object bringing over only the fields allowed in the Knative API. This
+// does not validate the contents or bounds of the provided fields.
+func PodSecurityContextMask(ctx context.Context, in *corev1.PodSecurityContext) *corev1.PodSecurityContext {
+	if in == nil {
+		return nil
+	}
+
+	out := new(corev1.PodSecurityContext)
+
+	if config.FromContextOrDefaults(ctx).Features.PodSpecSecurityContext == config.Disabled {
+		return out
+	}
+
+	out.RunAsUser = in.RunAsUser
```

Comment:
so one thing I feel a little weird about here that feels worth discussing. This PR ends up not just exposing some new fields on the existing userContainer.SecurityContext (that bit is totally fine), it also exposes the _pod_ security context to the user. Previously the user couldn't touch the pod's security spec, only the spec of their own containers. 

I see why we need to do this -- the fsGroup is only present at the pod level, because volumes are shared -- but it seems like quite a big new bit of surface area. For example it means the Queue proxy must now run in whatever podSecurityContext the user supplies (modulo fields that it can explicitly override in the container securityContext).

Few potential upshots of that, off the top of my head:
 - this will affect the ownership of the log volumes we mount in to queue proxy (it will arbitrarily chown them, I think..). 
 - any additional sidecars that are injected will also inherit these settings, unless overridden (e.g. Istio in the mesh case, I think). 
  - I've had a sneaky idea in the back of my head for a while that it'd be pretty nice to have a custom Queue Proxy that uses cgroups to pause/unpause the user container when there are no requests, for a lambda-like UX -- that likely requires giving QP elevated permissions. (etc)

Anyway, this may well be fine and reasonable, but I wanted to flag this up and make sure we explicitly think it through and decide it's ok.

Reply:
>  (modulo fields that it can explicitly override in the container securityContext).

yeah for now that being the `fsGroup` property 

> this will affect the ownership of the log volumes we mount in to queue proxy (it will arbitrarily chown them, I think..).

This makes me think there is an existing issue with our `/var/log` logging. ie. user container writes logs to an emptyDir as root (unsure if uid mapping get applied). If the daemon set logging agent was running as non-root it could have permission problems reading the logs. Don't know the k8s internals deep enough.

> Anyway, this may well be fine and reasonable, but I wanted to flag this up and make sure we explicitly think it through and decide it's ok.

I think this discussion is good and surfaces some potential upstream requirements in K8s (ie. does fsGroup property on the container security context make sense). For some existing pod-spec feature flags I don't think we intent to every change the default from `Disabled` - this flag may fall into the same class. Thus we should aim to at least let document to operators the issues you've surfaced.

This reminds me to update the `_example` key of the config map